### ISSUE TITLE
Basic Reflection

### ISSUE THREAD

**Issue by geoffreymcgill** [bridgedotnet/Bridge#15]
Related forum threads, issues and other:
http://forums.bridge.net/forum/community/help/1849
http://forums.bridge.net/forum/community/help/2168
http://forums.bridge.net/forum/community/help/2250
#1277 
#715

**Comment by michaelcheers** [comment]
I like the current way Bridge is implemented now with lots of Template (s) and External (s) with existing javascript classes. I like this implementation and don't want to lose it in Bridge 2.0. So, I think this is the way it should work:
you should have it generate 2 javascript files by default one with reflection details and one without,
classes should have an option even if marked external whether to add reflection or not.
In bridge.js you would probably have an option for the default:
```json
{
     //Uncomment this to always generate reflection.
     //The default is only on non-external classes.
     //"generateReflection":"always",

     // Uncomment this to never generate reflection.
     // The default is only on non-external classes.
     //"generateReflection":"never"
}
```
So, a class with Template (s) would have them turned into reflection and not be **just** Template (s).
When reflection is needed I think that it should a call something inside bridge.js and load all of the reflection files. So, if your project was named "BridgeTest" then you would have:
bridgeTest.reflection.js
bridge.reflection.js.
You could also have an option whether to embed reflection or not so it wouldn't need to load other javascript files. Another cool option would be to use .reflection.min.js files instead of .min.js files. So, you didn't need to always load the uncompressed ones.

**Comment by geoffreymcgill** [comment]
Hi @michaelcheers - Thanks for the comments and feature request(s). We're working on Reflection support right now, and we'll review your request in detail.

**Comment by geoffreymcgill** [comment]
This should be closed soon. We are including this functionality in Bridge 15.

**Comment by geoffreymcgill** [comment]
Reflection support is included Bridge as of version 15.0.
